src/services/chains/index.test.ts

```diff
@@ -957,4 +871,47 @@ function getTransactionDetailTestCases() {
   //   assert.equal(transactionInfo.currency, 'HNY');
   //   assert.equal(transactionInfo.amount, amount);
   // });
+
+  /// SOLANA
+  it('should return transaction detail for SOL transfer on Solana ', async () => {
+    // https://explorer.solana.com/tx/5GQGAgGfMNypB5GN4Pp2t3mEMky89bbpZwNDaDh1LJXopVm3bgSxFUgEJ4tEjf2NdibxX4NiiA752Ya2hzg2nqj8?cluster=devnet
+    const amount = 0.001;
+    const transactionInfo = await getTransactionInfoFromNetwork({
+      txHash:
+        '5GQGAgGfMNypB5GN4Pp2t3mEMky89bbpZwNDaDh1LJXopVm3bgSxFUgEJ4tEjf2NdibxX4NiiA752Ya2hzg2nqj8',
+      symbol: 'SOL',
+      chainType: ChainType.SOLANA,
+      networkId: NETWORK_IDS.SOLANA,
+      fromAddress: '5GECDSGSWmMuw6nMfmdBLapa91ZHDZeHqRP1fqvQokjY',
+      toAddress: 'DvWdrYYkwyM9mnTetpr3HBHUBKZ22QdbFEXQ8oquE7Zb',
+      timestamp: 1702931400,
+      amount,
+    });
+    assert.isOk(transactionInfo);
+    assert.equal(transactionInfo.currency, 'SOL');
+    assert.equal(transactionInfo.amount, amount);
+  });
+
+  it('should return error when transaction time is newer than sent timestamp for SOL transfer on Solana', async () => {
+    // https://explorer.solana.com/tx/5GQGAgGfMNypB5GN4Pp2t3mEMky89bbpZwNDaDh1LJXopVm3bgSxFUgEJ4tEjf2NdibxX4NiiA752Ya2hzg2nqj8?cluster=devnet
+
+    const amount = 0.001;
+    const badFunc = async () => {
+      await getTransactionInfoFromNetwork({
+        txHash:
+          '5GQGAgGfMNypB5GN4Pp2t3mEMky89bbpZwNDaDh1LJXopVm3bgSxFUgEJ4tEjf2NdibxX4NiiA752Ya2hzg2nqj8',
+        symbol: 'SOL',
+        chainType: ChainType.SOLANA,
+        networkId: NETWORK_IDS.SOLANA,
+        fromAddress: '5GECDSGSWmMuw6nMfmdBLapa91ZHDZeHqRP1fqvQokjY',
+        toAddress: 'DvWdrYYkwyM9mnTetpr3HBHUBKZ22QdbFEXQ8oquE7Zb',
+        timestamp: 1702931400 + ONE_DAY,
+        amount,
+      });
+    };
+    await assertThrowsAsync(
+      badFunc,
+      errorMessages.TRANSACTION_CANT_BE_OLDER_THAN_DONATION,
+    );
+  });
```

Comment:
Please add more tests if you can to cover other cases. You can use other transactions, like
3nzHwgxAu7mKw1dhGTVmqzY8Yet3kGWWqP5kr5D2fw1HzqPjqDGDe6xT5PguKXk8nAJcK4GpBEKWw7EzoLykKkCx

Reply:
@aminlatifi 
I just added another successful test case with your tx, for adding more test cases to cover all scenarios I need more time, does it needed for now or I can do that later?